classes/middleware/HasRoles.php

```diff
@@ -45,18 +47,29 @@ class HasRoles
     public function handle(Request $request, Closure $next, string $matchableRoles, int $rolesMatchingCriteria = HasRoles::ROLES_MATCH_LOOSE)
     {
         $user = $request->user(); /** @var \PKP\user\User $user */
```

Comment:
I didn't check how this middleware is handled, but the request might not have a user, so it sounds reasonable to include an extra check.

Reply:
Probably not a bad idea, however the `HasRole` middleware should always followed by a `HasUser` middleware . But as the HasUser middleware may use as stand alone and there is no such easy way to define that `HasRole` must follow the `HasUser`, it does leave a very small chance misuse that will cause exception . 

My suggestion is that rather than a check, probably an exception is better if there is no user . Basically without a user `HasRole` does not seems to have any meaning .

Reply:
Should I add such exception to this PR?

Reply:
If there's a chance of misuse, then go ahead with a good exception message regarding what Touhidur wrote 😁

Reply:
added exception https://github.com/pkp/pkp-lib/pull/10267/commits/11013232e58314d1957d4f2d2991feaf0ac59d35